pkg/services/certgenerator/certutil.go

```diff
@@ -0,0 +1,384 @@
+package certgenerator
+
+import (
+	"bytes"
+	cryptorand "crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"math/big"
+	"os"
+	"time"
+
+	certutil "k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/keyutil"
+	"k8s.io/kubernetes/pkg/controlplane"
+	netutils "k8s.io/utils/net"
+	"net"
+	"strings"
+)
+
+const (
+	ApiServerCertSerial = iota + 1
```

Comment:
The CA gets made using a helper in `cert` package provided in `client-go`. As such, it picks `0` for the CA serial - hence starting at 1 here.

Reply:
```suggestion
	// The CA gets made using a helper in cert package provided in client-go. As such, it picks 0 for the CA serial - hence starting at 1 here.
	ApiServerCertSerial = iota + 1
```